.github/workflows/pr.yml

```diff
@@ -521,93 +522,26 @@ jobs:
-          restore-keys: sbt
-      - uses: actions/download-artifact@v4
-        with:
-          name: build-target
-      - name: Untar artifacts
-        shell: bash
-        run:  tar xfz target.tgz
-      - uses: actions/download-artifact@v4
-        with:
-          name: build-fe-dist
-      - name: Untar fe artifacts
-        shell: bash
-        run:  tar xfz fe-dist.tgz
-      - name: Login to Docker Hub
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.DOCKERHUB_USER }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Setup buildx builder
-        uses: docker/setup-buildx-action@v3
-        with:
-          platforms: linux/amd64,linux/arm64
-          config-inline: |
-            [worker.oci]
-              max-parallelism = 1
-      - name: Try to free space
-        run: |
-          # Workaround to prevent "no space left on device" during buildx build - https://github.com/actions/runner-images/issues/2840#issuecomment-790492173
-          sudo rm -rf /usr/share/dotnet
-          sudo rm -rf /opt/ghc
-          sudo rm -rf "/usr/local/share/boost"
-          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
-          sudo rm -rf /usr/local/lib/android
-      - uses: sbt/setup-sbt@v1
-      - name: Publish with scaladocs
-        if: ${{ github.ref == 'refs/heads/staging' }}
-        env:
-          # Workaround to preveent "no space left on device" during publishDocker failed to create temp dir: mkdir /tmp/containerd-mountXYZ
-          TMPDIR: ${{ github.workspace }}
-        shell: bash
-        #TODO: handle version better, do we want to publish docker image for older scala versions? If so, how should it be tagged?
-        run: sbt "set ThisBuild / version := \"$NUSSKNACKER_VERSION\"" +publish +dist/Docker/publish +liteEngineRuntimeApp/Docker/publish
-      - name: Publish without scaladocs
-        if: ${{ github.ref != 'refs/heads/staging' }}
-        env:
-          # Workaround to preveent "no space left on device" during publishDocker failed to create temp dir: mkdir /tmp/containerd-mountXYZ
-          TMPDIR: ${{ github.workspace }}
-        shell: bash
-        #TODO: handle version better, do we want to publish docker image for older scala versions? If so, how should it be tagged?
-        run: sbt "set ThisBuild / version := \"$NUSSKNACKER_VERSION\"; set ThisBuild / packageDoc / publishArtifact := false; set Compile / doc / sources := Seq.empty" +publish +dist/Docker/publish +liteEngineRuntimeApp/Docker/publish
+
+  publish-after-tests:
+    name: Publish With Tests
+    uses: ./.github/workflows/shared/publish.yml
+    needs: ['build', 'build-fe', 'setup', 'tests', 'integrationTests', 'slowTests', 'frontendTests', 'cypressTests']
+    with:
+      ref: ${{ github.ref }}
+    if: ${{ github.ref != 'refs/heads/staging' }}
+
+  publish-before-tests:
+    name: Publish Without Tests
+    uses: ./.github/workflows/shared/publish.yml
+    needs: ['build', 'build-fe', 'setup']
```

Comment:
Just a note for purpose of further works: we should consider removing only `cypressTests` instead of all tests steps. Let's ignore this comment for now and check how this approach will work.